### PR TITLE
Add IPoolFactory interface; Add collateralization test checks to purchaseBid

### PR DESCRIPTION
**Pull request by MikeHathaway**

### CODE REVIEW

**Comment by MikeHathaway** [comment]
There seems to be an issue with the nightly build, investigating now. The tests run fine locally.

**Comment by MikeHathaway** [comment]
> There seems to be an issue with the nightly build, investigating now. The tests run fine locally.

DS-Test was pulled directly into Forge and no longer needs to be imported directly. I've removed the submodule accordingly.

File: src/_test/Multicall.t.sol [hunk 1]
```diff
@@ -16,17 +16,18 @@ contract MulticallTest is DSTestPlus {
...
-    ERC20PoolFactory internal _factory;
     PositionManager  internal _positionManager;
     QuoteToken       internal _quote;
 
     function setUp() external {
-        _collateral      = new CollateralToken();
-        _quote           = new QuoteToken();
-        _pool            = new ERC20PoolFactory().deployPool(address(_collateral), address(_quote));
-        _positionManager = new PositionManager();
+        _collateral             = new CollateralToken();
```
Comment (grandizzy) [inline src/_test/Multicall.t.sol]: there are extra spaces here

File: src/_test/ERC20PoolCollateral.t.sol [hunk 2]
```diff
@@ -15,19 +15,22 @@ import { UserWithCollateral, UserWithQuoteToken } from "./utils/Users.sol";
...
-    UserWithCollateral internal _borrower;
-    UserWithCollateral internal _borrower2;
-    UserWithQuoteToken internal _lender;
-    UserWithQuoteToken internal _lender1;
-    UserWithCollateral internal _bidder;
+    address             internal _poolAddress;
+    CollateralToken     internal _collateral;
+    ERC20Pool           internal _pool;
+    QuoteToken          internal _quote;
+    UserWithCollateral  internal _borrower;
```
Comment (grandizzy) [inline src/_test/ERC20PoolCollateral.t.sol]: rm extra space between `UserWithCollateral` and `internal`

File: src/_test/ERC20PoolBorrow.t.sol [hunk 3]
```diff
@@ -15,17 +15,20 @@ import { UserWithCollateral, UserWithQuoteToken } from "./utils/Users.sol";
...
-    ERC20Pool          internal _pool;
-    QuoteToken         internal _quote;
-    UserWithCollateral internal _borrower;
-    UserWithCollateral internal _borrower2;
-    UserWithQuoteToken internal _lender;
+    address             internal _poolAddress;
+    CollateralToken     internal _collateral;
+    ERC20Pool           internal _pool;
+    QuoteToken          internal _quote;
+    UserWithCollateral  internal _borrower;
```
Comment (grandizzy) [inline src/_test/ERC20PoolBorrow.t.sol]: rm extra space between `UserWithCollateral` and `internal`

File: src/_test/ERC20PoolBid.t.sol [hunk 4]
```diff
@@ -47,6 +51,8 @@ contract ERC20PoolBidTest is DSTestPlus {
         _lender.addQuoteToken(_pool, address(_lender), 3_000 * 1e18, _p3010);
         _lender.addQuoteToken(_pool, address(_lender), 3_000 * 1e18, _p1004);
         assertEq(_pool.totalQuoteToken(), 9_000 * 1e45);
+        assertEq(_pool.getPoolCollateralization(), Maths.ONE_RAY);
```
Comment (grandizzy) [inline src/_test/ERC20PoolBid.t.sol:54]: allign line 53 `9_000 * 1e45` to the new tests

**Review by grandizzy — CHANGES_REQUESTED** [review]
LGTM, only minor changes with code format

File: src/_test/ERC20PoolBid.t.sol (hunk 4 above)
Comment (MikeHathaway) [reply src/_test/ERC20PoolBid.t.sol:54]: Updated

File: src/_test/ERC20PoolBorrow.t.sol (hunk 3 above)
Comment (MikeHathaway) [reply src/_test/ERC20PoolBorrow.t.sol]: removed

File: src/_test/ERC20PoolCollateral.t.sol (hunk 2 above)
Comment (MikeHathaway) [reply src/_test/ERC20PoolCollateral.t.sol]: removed

File: src/_test/Multicall.t.sol (hunk 1 above)
Comment (MikeHathaway) [reply src/_test/Multicall.t.sol]: removed

**Review by grandizzy — APPROVED** [review]

**Pull request merged**